### PR TITLE
Api to render content to html

### PR DESCRIPTION
**Pull request by monassfaro**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    private final TenantContextHolder tenantContextHolder;
+    private final Configuration freeMarkerConfiguration;
+
+    @SneakyThrows
+    public RenderTemplateResponse renderEmailContent(RenderTemplateRequest renderTemplateRequest) {
+        try {
+            String tenantKey = tenantContextHolder.getTenantKey();
+            String templateKey = format("%s/en/%s", tenantKey, UUID.randomUUID());
+            Template mailTemplate = new Template(templateKey, renderTemplateRequest.getContent(), freeMarkerConfiguration);
```
[the remark this answers is not in the document]
Comment (monassfaro) [reply src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java]: Replaced templateKey with UUID

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        try {
+            Template mailTemplate = new Template(UUID.randomUUID().toString(), renderTemplateRequest.getContent(), freeMarkerConfiguration);
+            String renderedContent = FreeMarkerTemplateUtils.processTemplateIntoString(mailTemplate, renderTemplateRequest.getModel());
+            RenderTemplateResponse renderTemplateResponse = new RenderTemplateResponse();
+            renderTemplateResponse.setContent(renderedContent);
+            return renderTemplateResponse;
+        } catch (TemplateException | IOException e) {
+            log.error("Template could not be rendered with content: {} and model: {}. Error: {}", renderTemplateRequest.getContent(),
+                renderTemplateRequest.getModel(), e.getMessage());
+            throw new RenderTemplateException(e.getMessage(), renderTemplateRequest.getContent(), renderTemplateRequest.getModel());
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java:40]: for template exception we need getMessageWithoutStackTop

File: src/main/java/com/icthh/xm/tmf/ms/communication/web/rest/errors/RenderTemplateException.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.icthh.xm.tmf.ms.communication.web.rest.errors;
+
+import lombok.Getter;
+import org.zalando.problem.AbstractThrowableProblem;
+
+import java.util.Map;
+
+@Getter
+public class RenderTemplateException extends AbstractThrowableProblem {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/web/rest/errors/RenderTemplateException.java:9]: RequiredArgumentException

**Review by sergeysenja1992 — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/communication/web/rest/EmailTemplateController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/templates")
+public class EmailTemplateController {
+
+    private final EmailTemplateService emailTemplateService;
+
+    @Timed
+    @PostMapping("/render")
+    public ResponseEntity<RenderTemplateResponse> renderEmailContentToHtml(@Valid @RequestBody RenderTemplateRequest renderTemplateRequest) {
```
[the remark this answers is not in the document]
Comment (monassfaro) [reply src/main/java/com/icthh/xm/tmf/ms/communication/web/rest/EmailTemplateController.java]: Fixed

File: src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java (hunk 2 above)
Comment (monassfaro) [reply src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java:44]: Fixed

**Review by YRedko — APPROVED** [review]

**Pull request merged**